clang/lib/CodeGen/CodeGenFunction.h

```diff
@@ -1411,6 +1411,8 @@ class CodeGenFunction : public CodeGenTypeCache {
   /// decls.
   DeclMapTy LocalDeclMap;
 
+  llvm::SmallDenseMap<const Expr *, LValue, 32> ExprLValueMap;
```

Comment:
Perhaps a comment as to what this map is used for?

Reply:
Done.

Reply:
Wont this store all kinds of unrelated Exprs? Is there any way we can minimize the size of this map to avoid storing things we wont need?

Reply:
It will. I did a test to see how many Exprs are typically stored. For both Linux and Clang, the vast majority of functions have fewer than 14 entries. And most (over 2 million for LLVM and over 90,000 for Linux) have no entries at all. It cuts off pretty drastically around 40 entries for each. The larger the number of entries the fewer the number of functions that have them. Both LLVM and Linux have a function that has 5,900 and 1,900 resp.

Reply:
I do this only when the sanitizer is enabled to keep us from building it unnecessarily.